process/factory/shard/vmContainerFactory.go

```diff
@@ -89,9 +94,28 @@ func (vmf *vmContainerFactory) Create() (process.VirtualMachinesContainer, error
 		return nil, err
 	}
 
+	// The vmContainerFactory keeps references to all containers it has created,
+	// in order to replace, from within the containers, those VM instances that
+	// become out-of-date after specific epochs.
+	vmf.containers = append(vmf.containers, container)
+
 	return container, nil
 }
 
+// EpochConfirmed updates VMs in all created containers depending on the epoch
+func (vmf *vmContainerFactory) EpochConfirmed(epoch uint32) {
+	if epoch >= vmf.ArwenGasManagementRewriteEnableEpoch {
```

Comment:
please use a bool flag here and a log print (check other EpochConfirmed functions) 
Why do we keep recreating the arwen instance **in each epoch** ?

Reply:
This is a mistake. 

Changed the condition to `==`, because the replacement should only be triggered on the specific epochs. If `Create()` is called afterwards, the correct Arwen will be instantiated anyway.